stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/gallery/internal/AttachmentGalleryVideoPageFragment.kt

```diff
@@ -0,0 +1,263 @@
+            attrs = null
+        )
+    }
+
+    /**
+     * If the video has been prepared and the video player
+     * is ready for playback.
+     *
+     * VideoView does not expose state so we use these to track
+     * state externally.
+     */
+    private var playbackPrepared = false
+        set(value) {
+            field = value
+            if (playbackPrepared && playbackStartRequested) {
+                binding.thumbnailImageView.visibility = View.GONE
+                binding.playButtonCardView.visibility = View.GONE
+                binding.progressBar.visibility = View.GONE
+            } else {
+                binding.thumbnailImageView.visibility = View.VISIBLE
+                binding.playButtonCardView.visibility = View.VISIBLE
+                binding.progressBar.visibility = View.GONE
+            }
+        }
+
+    /**
+     * If the user has pressed play.
+     *
+     * VideoView does not expose state so we use these to track
+     * state externally.
+     */
+    private var playbackStartRequested = false
+        set(value) {
+            field = value
+            if (!playbackPrepared && playbackStartRequested) {
+                binding.progressBar.visibility = View.VISIBLE
+                binding.playButtonCardView.visibility = View.GONE
+                binding.thumbnailImageView.visibility = View.GONE
+            } else {
+                binding.playButtonCardView.visibility = View.GONE
+                binding.thumbnailImageView.visibility = View.GONE
+                binding.progressBar.visibility = View.GONE
+            }
+        }
+
+    /**
+     * Resets the state and hides the controller.
+     *
+     * Important for resetting state when paging through
+     * fragments in a pager.
+     */
+    override fun onPause() {
+        super.onPause()
+        resetState()
+        mediaController.hide()
+    }
+
+    /**
+     * Contains the URL to the thumbnail of the video.
+     */
+    private val thumbUrl: String? by lazy {
+        requireArguments().getString(ARG_THUMB_URL)
+    }
+
+    /**
+     * Contains the URL necessary to reproduce the video.
+     */
+    private val assetUrl: String? by lazy {
+        requireArguments().getString(ARG_ASSET_URL)
+    }
+
+    private val mediaController: MediaController by lazy {
+        createMediaController(requireContext())
+    }
+
+    private var imageClickListener: () -> Unit = {}
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return StreamUiItemAttachmentGalleryVideoBinding.inflate(inflater)
+            .apply { _binding = this }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupVideoThumbnail()
+        setupPlayButton()
+        loadVideo()
+    }
+
+    private fun setupVideoThumbnail() {
+        if (ChatUI.videoThumbnailsEnabled) {
```

Comment:
Not an issue, but feels a bit low level for `ChatUI`. Let's discuss later. Could be for exmaple:
```
ChatUI.meesageListConfig.videoThumbnailsEnabled
```